pyspades/master.py

```diff
@@ -96,13 +97,16 @@ def on_disconnect(self):
             self.defer = None
         callback = self.disconnect_callback
         if callback is not None:
-            callback()
+            callback(self)
         self.disconnect_callback = None
 
 
 def get_master_connection(protocol):
-    defer = Deferred()
-    connection = protocol.connect(MasterConnection, HOST, PORT, MASTER_VERSION)
-    connection.server_protocol = protocol
-    connection.defer = defer
-    return defer
+    defers = []
+    for host in HOSTS:
```

Comment:
Similarly to the last PR, There's an overall thing I'm trying to do, which is move all configuration out of the individual modules and into parameters passed in from `server.py`.

So here, I'd similarly prefer if instead of using global variables, the server name and port was instead passed in via parameters. This function would then return just one `Deferred`, and be called once per master server. This also allows us to create master connections with different protocol implementations once we implement extensions to that.